### PR TITLE
Updated README for surrealdb go driver

### PR DESCRIPTION
**Pull request by sbshah97**
This PR is to make README updates to ensure users are not confused by the starter code. I have tested this locally on my system.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,37 +17,39 @@ The official SurrealDB library for Golang.
...
 )
 
+type User struct {
+	ID      string `json:"id,omitempty"`
+	Name    string `json:"name"`
+	Surname string `json:"surname"`
+}
+
 func main() {
-	// Connect to SurrealDB
```
Comment (phughk) [inline README.md:31]: I can't apply a suggestion to a deleted line, but could we revert this please? We do actually have people who aren't familiar with connecting to databases, so it would be good to have the comment

File: README.md [hunk 2]
```diff
@@ -17,37 +17,39 @@ The official SurrealDB library for Golang.
 go get github.com/surrealdb/surrealdb.go
 ```
 
-### Usage
+### Quick Start
```
Comment (phughk) [inline README.md:20]:
```suggestion
### Usage
```

File: README.md [hunk 3]
```diff
@@ -17,37 +17,39 @@ The official SurrealDB library for Golang.
...
 		panic(err)
 	}
 
-	// Select namespace and database
 	if _, err = db.Use("test", "test"); err != nil {
 		panic(err)
 	}
 
-	// Create user struct
+	// Create user
```
Comment (phughk) [inline README.md:52]:
```suggestion
	// Define user struct
```

File: README.md [hunk 4]
```diff
@@ -81,16 +83,12 @@ func main() {
 
 	// Change part/parts of user
 	changes := map[string]string{"name": "Jane"}
-	if _, err = db.Change(selectedUser.ID, changes); err != nil {
```
Comment (phughk) [inline README.md:84]: Why was this removed?

File: README.md [hunk 5]
```diff
@@ -101,10 +99,17 @@ func main() {
 	if _, err = db.Delete(selectedUser.ID); err != nil {
 		panic(err)
 	}
-}
```
Comment (phughk) [inline README.md:104]: The spacing was preferable before, no?

File: README.md [hunk 6]
```diff
@@ -101,10 +99,17 @@ func main() {
...
+}
 ```
 
+* Step 1: Create a file called `main.go` and paste the above code
+* Step 2: Run the command `go mod init github.com/<github-username>/<project-name>` to create a go.mod file
+* Step 3: Run the command `go mod tidy` to download surreal db
+* Step 4: Run `go run main.go` to run the application. 
+
+Note: Would be helpful to add `fmt.Println` statements to debug.
+
```
Comment (phughk) [inline README.md:112]: Not sure if the note is necessary
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -17,37 +17,39 @@ The official SurrealDB library for Golang.
...
 func main() {
-	// Connect to SurrealDB
 	db, err := surrealdb.New("ws://localhost:8000/rpc")
 	if err != nil {
 		panic(err)
 	}
 
-	// Sign in
-	if _, err = db.Signin(map[string]string{
```
Comment (phughk) [inline README.md:38]: Please could you revert this to what it was before? We do want the comment for clarity, and we do want to be more specific with types if we can

**Review by phughk — COMMENTED** [review]
Leaving some comments but thank you for submitting this!

File: README.md (hunk 1 above)
Comment (sbshah97) [reply README.md:31]: Yep absolutely. The other thing is do you think it would make sense to have a line on how to install and start SurrealDB on your local system or atleast point to documentation which can help them figure it out.

Comment (phughk) [reply README.md:31]: Yes, a link is better thank you!

File: README.md (hunk 7 above)
Comment (sbshah97) [reply README.md:38]: Screenshot 2023-11-19 at 23 17 56

Makes sense!

File: README.md (hunk 4 above)
Comment (sbshah97) [reply README.md:84]: This was not working and giving a panic hence removed it. I can add it if we have figured out why it was not working.

File: README.md (hunk 1 above)
Comment (sbshah97) [reply README.md:31]: This is done above.

File: README.md (hunk 4 above)
Comment (ElecTwix) [reply README.md:84]: in #98 it has been changed to modify.
Try the main branch instead of the latest release.

File: README.md (hunk 7 above)
Comment (phughk) [reply README.md:38]: Ideally the fields would be known. In case people make mistakes like `username` instead of `user`, or `password` instead of `pass` or not be aware of the other fields. Overall, the parameter needs to change.

**Review by phughk — APPROVED** [review]
Lgtm thanks!
